internal/concierge/impersonator/impersonator.go

```diff
@@ -315,6 +322,31 @@ func newInternal( //nolint:funlen // yeah, it's kind of long.
 	return result, nil
 }
 
+func getReverseProxyClient(clientOpts []kubeclient.Option) (*kubeclient.Client, error) {
+	// just use the overrides given during unit tests
+	if len(clientOpts) != 0 {
+		return kubeclient.New(clientOpts...)
+	}
+
+	// this is the magic path where the impersonation proxy SA token is mounted
+	const tokenFile = "/var/run/secrets/impersonation-proxy.concierge.pinniped.dev/serviceaccount/token" //nolint:gosec // this is not a credential
```

Comment:
Nit: should this `pinniped.dev` be different if the API suffix is changed? I think I like it as-is but I thought I'd point that out as well.

Reply:
I don't think making it dynamic adds any value and would create a bunch of extra wiring.